### PR TITLE
Feat categorywise playlist

### PR DESCRIPTION
**Pull request by Kothalkarpriya**
-Added category
-Now, we can check category wise videos 
-Fixed styling

### CODE REVIEW

**Review by ajit-sharmaGH — APPROVED** [review]
Everything looks good 👍
You can merge it

File: src/components/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <div className="category-container">
+      <button
+        onClick={() => {
+          filterDispatch({ type: "SET_CATEGORY", payload: "ALL" });
+        }}
+      >
+        All
+      </button>
+      {loading ? (
+        <h4>Loading...</h4>
```
Comment (paarthbajaj) [inline src/components/Filter.jsx:21]: Implementing this at the start is really a good thing.

**Review by paarthbajaj — APPROVED** [review]
All looks good and clean. No suggestions. You can merge.

File: src/components/Filter.jsx (hunk 1 above)
Comment (Kothalkarpriya) [reply src/components/Filter.jsx:21]: Okay

File: src/assests/styles/video-listing.css [hunk 2]
```diff
@@ -19,11 +20,10 @@
     width: 250px;
```
Comment (t007rushi) [inline src/assests/styles/video-listing.css:20]: repeated width prop?

File: src/backend/utils/filterReducer.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+function filterReducer(state, action) {
+  switch (action.type) {
+    case "RESET":
+      return { ...state, filterSelect: action.payload };
+    case "SET_CATEGORY":
+      return { ...state, filterSelect: action.payload };
```
Comment (t007rushi) [inline src/backend/utils/filterReducer.jsx:6]: both looks doing same actions can make single UPDATE case

**Review by t007rushi — COMMENTED** [review]
Look for minor changes

**Pull request merged**